src/tsserver/server.ts

```diff
@@ -648,374 +27,75 @@ namespace ts.server {
-
-    function parseStringArray(argName: string): readonly string[] {
-        const arg = findArgument(argName);
-        if (arg === undefined) {
-            return emptyArray;
-        }
-        return arg.split(",").filter(name => name !== "");
-    }
-
-    let unknownServerMode: string | undefined;
-    function parseServerMode(): LanguageServiceMode | undefined {
-        const mode = findArgument("--serverMode");
-        if (mode === undefined) {
-            return undefined;
-        }
-
-        switch (mode.toLowerCase()) {
-            case "semantic":
-                return LanguageServiceMode.Semantic;
-            case "partialsemantic":
-                return LanguageServiceMode.PartialSemantic;
-            case "syntactic":
-                return LanguageServiceMode.Syntactic;
-            default:
-                unknownServerMode = mode;
-                return undefined;
-        }
-    }
-
-    const globalPlugins = parseStringArray("--globalPlugins");
-    const pluginProbeLocations = parseStringArray("--pluginProbeLocations");
-    const allowLocalPluginLoads = hasArgument("--allowLocalPluginLoads");
-
-    const useSingleInferredProject = hasArgument("--useSingleInferredProject");
-    const useInferredProjectPerProjectRoot = hasArgument("--useInferredProjectPerProjectRoot");
-    const disableAutomaticTypingAcquisition = hasArgument("--disableAutomaticTypingAcquisition");
-    const suppressDiagnosticEvents = hasArgument("--suppressDiagnosticEvents");
-    const syntaxOnly = hasArgument("--syntaxOnly");
-    const serverMode = parseServerMode();
-    const telemetryEnabled = hasArgument(Arguments.EnableTelemetry);
-    const noGetErrOnBackgroundUpdate = hasArgument("--noGetErrOnBackgroundUpdate");
-
-    const commandLineTraceDir = findArgument("--traceDirectory");
-    const traceDir = commandLineTraceDir
-        ? stripQuotes(commandLineTraceDir)
-        : process.env.TSS_TRACE;
-    if (traceDir) {
-        tracing.startTracing(tracing.Mode.Server, traceDir);
-    }
-
-    logger.info(`Starting TS Server`);
-    logger.info(`Version: ${version}`);
-    logger.info(`Arguments: ${process.argv.join(" ")}`);
-    logger.info(`Platform: ${os.platform()} NodeVersion: ${nodeVersion} CaseSensitive: ${sys.useCaseSensitiveFileNames}`);
-    logger.info(`ServerMode: ${serverMode} syntaxOnly: ${syntaxOnly} hasUnknownServerMode: ${unknownServerMode}`);
-
-    const ioSession = new IOSession();
-    process.on("uncaughtException", err => {
-        ioSession.logError(err, "unknown");
-    });
-    // See https://github.com/Microsoft/TypeScript/issues/11348
-    (process as any).noAsar = true;
-    // Start listening
-    ioSession.listen();
-
-    if (Debug.isDebugging) {
-        Debug.enableDebugInfo();
-    }
-
-    if (ts.sys.tryEnableSourceMapsForHost && /^development$/i.test(ts.sys.getEnvironmentVariable("NODE_ENV"))) {
-        ts.sys.tryEnableSourceMapsForHost();
+    switch (systemKind) {

```

Comment:
The enum infrastructure seems unnecessary.  It looks like this could just be 
```ts
if (!!process) {
    start(initializeNodeSystem);
}
else {
    // Web
}
```